src/metatrain/utils/data/target_info.py

```diff
@@ -212,13 +230,33 @@ def is_compatible_with(self, other: "TargetInfo") -> bool:
             other_block = other.layout[key]
             if not block.samples == other_block.samples:
                 return False
-            if not block.components == other_block.components:
+            if not _is_labelslist_equal(block.components, other_block.components):
                 return False
             if not block.properties == other_block.properties:
                 return False
             # gradients are not checked on purpose
         return True
 
+    def to(self, device: torch.device) -> "TargetInfo":
+        """Move the target info to the given device."""
+        new_layout = self.layout.to(device=device)
```

Comment:
Is there also a `dtype` one could change?

Reply:
Hmm yes this was one of my doubts, probably it's better to add it no?

Reply:
I would say so. Should be easy to add.